From 6655a7020bb7bee9d503618f626916da1db917e7 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Date: Sat, 26 Feb 2022 23:07:54 +0300
Subject: [PATCH 08/24] Bluetooth: hci_qca: reopen serial port after toggling
 power

Reopen the serial port after toggling the power. This saves us from
getting command timeouts on first command submitted.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
---
 drivers/bluetooth/hci_qca.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/bluetooth/hci_qca.c b/drivers/bluetooth/hci_qca.c
index e30c97953..0c2a182be 100644
--- a/drivers/bluetooth/hci_qca.c
+++ b/drivers/bluetooth/hci_qca.c
@@ -1699,6 +1699,8 @@ static int qca_power_on(struct hci_dev *hdev)
 			gpiod_set_value_cansleep(qcadev->bt_en, 1);
 			/* Controller needs time to bootup. */
 			msleep(150);
+			serdev_device_close(hu->serdev);
+			ret = serdev_device_open(hu->serdev);
 		}
 	}
 
-- 
2.34.1

